File: Proyecto en C#/Entrega Diseno/Jubilado.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Entrega_Diseno
{
    class Jubilado: CategoriaPersona
    {
        private double _PrecioDescuento;
        public double PrecioDescuento
        {
            get { return _PrecioDescuento; }
            set { _PrecioDescuento = 0.85 ; }
        }

        float CategoriaPersona.aplicarDescuento(float entradaBase)
        {
            return ( (float)(entradaBase * _PrecioDescuento));
        }
    }
}

File: src/src/Jubilado.scala
class Jubilado (PrecioDescuento: Double) extends CategoriaPersona{

   var precioDescuento: Double = PrecioDescuento
   
   override def aplicarDescuento(EntradaBase: Double):Double = {
     return (EntradaBase * this.precioDescuento)
   }
}